### PR TITLE
[RISCV][GISel] Remove source constraint from selectCopy and use it fo…

### PR DESCRIPTION
**Pull request by topperc**
…r G_ANYEXT/G_TRUNC.

AArch64, Mips, and ARM do not have source constraints in their version of selectCopy. I'm assuming we don't need it either.

We weren't constraining the destination register of G_ANYEXT/G_TRUNC previously, but we got away with it in our tests because all their users constrained them as a use. When I removed the source constraint from selectCopy this stopped working.

### CODE REVIEW

**Comment by michaelmaitland** [comment]
Test case?

File: llvm/lib/Target/RISCV/GISel/RISCVInstructionSelector.cpp [hunk 1]
```diff
@@ -271,37 +270,25 @@ const TargetRegisterClass *RISCVInstructionSelector::getRegClassForTypeOnBank(
 bool RISCVInstructionSelector::selectCopy(MachineInstr &MI,
                                           MachineRegisterInfo &MRI) const {
   Register DstReg = MI.getOperand(0).getReg();
-  Register SrcReg = MI.getOperand(1).getReg();
 
-  if (Register::isPhysicalRegister(SrcReg) &&
-      Register::isPhysicalRegister(DstReg))
+  if (DstReg.isPhysical())
```
Comment (michaelmaitland) [inline llvm/lib/Target/RISCV/GISel/RISCVInstructionSelector.cpp:274]: Are the select functions supposed to return `true` if a change was made?

**Comment by topperc** [comment]
> Test case?

I don't have a test case because it wasn't technically wrong. It was different than other targets and I don't know why.

File: llvm/lib/Target/RISCV/GISel/RISCVInstructionSelector.cpp (hunk 1 above)
Comment (topperc) [reply llvm/lib/Target/RISCV/GISel/RISCVInstructionSelector.cpp:274]: I believe they are supposed to return true for success and false for failure.

**Review by michaelmaitland — APPROVED** [review]
LGTM.

**Pull request merged**